loosen assertion on category size to allow for aligment padding

<rdar://problem/10272666>

--- src/ld/passes/objc.cpp
+++ src/ld/passes/objc.cpp
@@ -813,7 +813,7 @@ void OptimizeCategories<A>::doit(const Options& opts, ld::Internal& state)
 					continue;
 				}
 				assert(categoryAtom != NULL);
-				assert(categoryAtom->size() == Category<A>::size());
+				assert(categoryAtom->size() >= Category<A>::size());
 				// ignore categories also in __objc_nlcatlist
 				if ( nlcatListAtoms.count(categoryAtom) != 0 )
 					continue;
